src/Microsoft.AspNetCore.OData/Adapters/WebApiContext.cs

```diff
@@ -100,6 +126,31 @@ public SelectExpandClause SelectExpandClause
             get { return this.innerFeature.TotalCount; }
         }
 
+        /// <summary>
+        /// Page size to be used by skiptoken implementation for the top-level resource for the request. 
+        /// </summary>
+        public int PageSize
+        {
+            get
+            {
+                ODataFeature feature = this.innerFeature as ODataFeature;
+                if (feature != null)
+                {
+                    return feature.PageSize;
+                }
+
+                return 0;
+            }
+            set
+            {
+                ODataFeature feature = this.innerFeature as ODataFeature;
+                if (feature != null)
```

Comment:
>feature != null [](start = 20, length = 15)

same as above; if innerFeature is not ODataFeature, should we set a private pagesize property?  In that case, would it get used for the skiptoken?

Reply:
The private field will be of no use as it will not persist when we serialize. 

---
In reply to: [293003110](https://github.com/OData/WebApi/pull/1724#discussion_r293003110) [](ancestors = 293003110)